enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,736 @@
+[crio.runtime.runtimes.runc]
+  runtime_path = "/usr/bin/runc"
+  runtime_type = "oci"
+  allowed_annotations = ["cpu.workload.openshift.io"]
+```
+
+Pods that have the `cpu.workload.openshift.io` annotation will have
+their cpuset configured to the value from the appropriate workload
+configuration, as well as have their CPU shares configured to the
+value of the annotation.
+
+Note that this field does not conflict with the `infra_ctr_cpuset`
+config option, as the infra container will still be put in that
+cpuset.  They can be configured as the same value if the infra
+container should also be considered to be managed.
+
+#### API Server Admission Hook
+
+A new admission hook in the kubernetes API server within OpenShift
+will mutate pods when they are created to make 2 changes.
+
+1. It will move CPU requests to workload CPU requests, so that
+   the scheduler can successfully place the pod on the node, even
+   though the sum of the CPU requests for all management workloads may
+   exceed the actual CPU capacity of the management CPU pool.
+2. It will add annotations with for each container with the value
+   equal to the original CPU requests, so that CRI-O can use the value
+   to configure the CPU shares for the container.
+
+We will not change pods in a way that changes their quality-of-service
+class. So, we would not strip CPU requests unless they also have
+memory requests, because if we mutate the pod so that it has no CPU or
+memory requests the quality-of-service class of the pod would be
+changed automatically. Any labeled pod that is already BestEffort
+would be annotated using `0` as the value so that CRI-O will have an
+indicator to configure the CPU shares as BestEffort.
+
+#### Kubelet Changes
+
+Kubelet will be changed to look for a configuration file,
+`/etc/kubernetes/management-pinning`, to enable the management
+workload partitioning feature. The file should contain the cpuset
+specifier for the CPUs making up the management CPU pool.
+
+Kubelet will be changed so that when the feature is enabled, [when a
+static pod definition is
+read](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/config/file_linux.go)
+it is mutated in a way similar to the API server admission hook will
+mutate regular pods.
+
+Kubelet will be changed so that when the feature is enabled, it
+advertises a new extended resource of
+`cpu.workload.openshift.io/management`, representing all of the CPU
+capacity of the host (not just the management CPU pool).
+
+#### Installer Changes
+
+The installer will be changed to accept a new `workload.partitions`
+parameter for each [machine
+pool](https://github.com/openshift/installer/blob/master/docs/user/customization.md#machine-pools)
+with a list of workload types and their settings. For now, the only
+setting will be the name and the CPU sets to treat as separate from
+the standard shared pool. The name `workload.partitions` may change,
+based on discussion on the installer code changes.
```

Comment:
Final name needs to be in the enhancement (as I mentioned before, we are adding a top level concept to openshift, that comes with a high expectation of due dilligence on naming)